Getting the codebase adapted to updates in APIs such as `GrantPermissionUi()` and `deleteFile()`

## Describe the Bug

Currently, the MovieStar codebase still uses older versions of `GrantPermissionUi()` and `deleteFile()`, which causes errors during compilation and runtime.

```
lib/utils/create_app_folder.dart:104:23: Error: Too many positional arguments: 0 allowed, but 1 found.
Try removing the extra positional arguments.
      await deleteFile('$basePath/$folderName');
                      ^
../../../../../.pub-cache/git/solidpod-43c271327b8b5e3ad74a7f1fd5c253db0b9bff02/lib/src/solid/utils/delete_helper.dart:216:14: Context: Found this candidate, but the arguments don't match.
Future<void> deleteFile({
             ^^^^^^^^^^
lib/screens/files_screen.dart:70:17: Error: No named parameter with the name 'basePath'.
                basePath: basePath,
                ^^^^^^^^
../../../../../.pub-cache/git/solidui-033a387c65bd9554171374f67f8c529a3c6534fa/lib/src/widgets/solid_file.dart:144:9: Context: Found this candidate, but the arguments don't match.
  const SolidFile({
        ^^^^^^^^^
lib/core/services/pod/file_operations_service.dart:259:32: Error: Too many positional arguments: 0 allowed, but 1 found.
Try removing the extra positional arguments.
      await solidpod.deleteFile(fileName);
                               ^
../../../../../.pub-cache/git/solidpod-43c271327b8b5e3ad74a7f1fd5c253db0b9bff02/lib/src/solid/utils/delete_helper.dart:216:14: Context: Found this candidate, but the arguments don't match.
Future<void> deleteFile({
             ^^^^^^^^^^
lib/widgets/sharing/display_widgets.dart:319:17: Error: Method not found: 'GrantPermissionUi'.
          body: GrantPermissionUi(
                ^^^^^^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception
Failed to package /Users/intony/Documents/Software Innovation Institute/01-Projects/07-Pods/moviestar.
Command PhaseScriptExecution failed with a nonzero exit code
** BUILD FAILED **

Error: Build process failed
```

Update the code to align with the latest API changes.

## Expected Behaviour

The codebase should use the latest APIs and could build and run with no errors.

## Context

Where has the issue been observed:

- [ ] Android
- [ ] Chrome
- [ ] iOS
- [ ] Linux
- [x] macOS
- [ ] Web
- [ ] Windows

## Closing Criteria

Checklist for closing the issue:

- [ ] No errors from `make prep`
- [ ] All tests pass `make qtest`
- [ ] The app could build and run with no errors.

--- a/lib/core/services/pod/file_operations_service.dart
+++ b/lib/core/services/pod/file_operations_service.dart
@@ -256,7 +256,7 @@ class PodFileOperationsService {
 
       // Use SolidPOD's deleteFile function directly.
 
-      await solidpod.deleteFile(fileName);
+      await solidpod.deleteFile(fileUrl: fileName);
 
       return PodFileOperationResult.success();
     } catch (e) {

--- a/lib/screens/files_screen.dart
+++ b/lib/screens/files_screen.dart
@@ -67,7 +67,7 @@ class _FilesScreenState extends State<FilesScreen> with ScreenStateMixin {
             return const Padding(
               padding: EdgeInsets.all(Dimensions.xl),
               child: SolidFile(
-                basePath: basePath,
+                currentPath: basePath,
               ),
             );
           },

--- a/lib/utils/create_app_folder.dart
+++ b/lib/utils/create_app_folder.dart
@@ -101,7 +101,7 @@ Future<SolidFunctionCallStatus> createAppFolder({
 
       // Full path for deletion needs to include basePath (e.g. moviestar/data).
 
-      await deleteFile('$basePath/$folderName');
+      await deleteFile(fileUrl: '$basePath/$folderName');
     }
 
     if (!context.mounted) {

--- a/lib/widgets/sharing/display_widgets.dart
+++ b/lib/widgets/sharing/display_widgets.dart
@@ -26,7 +26,7 @@ library;
 import 'package:flutter/material.dart';
 
 import 'package:cached_network_image/cached_network_image.dart';
-import 'package:solidpod/solidpod.dart' show GrantPermissionUi;
+import 'package:solidui/solidui.dart' show GrantPermissionUi;
 
 import 'package:moviestar/widgets/sharing/components.dart';
 import 'package:moviestar/widgets/sharing/form_widgets.dart';

